node/network/gossip-support/src/lib.rs

```diff
@@ -64,6 +70,11 @@ struct State {
 	// at least a third of authorities the last time.
 	// `None` otherwise.
 	last_failure: Option<Instant>,
+
+	/// First time we did not reach our connectivity treshold.
+	///
+	/// Will be cleared once we reached it.
+	first_failure: Option<Instant>,
```

Comment:
Could we not use a `Vec` of actual errors and derive everything else from that. At least one could provide information on what went wrong as well / which authorities were not reachable
.

Reply:
There is only one kind of error - the lookup failed. Ideally we would have also track `PeerConnected` `PeerDisconnected` events and monitor real connectivity this way. I will create an issue for that. Thanks!

We don't know which authorities were not reachable at that point - we only get the number of failed lookups back. Listing all the validators would be very verbose and usually of little value anyways.

Reply:
Of it's only a 1bit result type, this indeed pointless